woocommerce/woocommerce
# Introduce asset size check action to CI

### Changes proposed in this Pull Request:

As part of a more focussed effort to improve performance and avoid accidentally negatively impacting bundle sizes, we need to ensure that each PR does not make any significant increase to the bundle size or at least inform the PR author so they can make an educated decision. 

This will not block PR merging, the idea is to help authors not make accidental detrimental changes to bundle size when they may not be aware of it.

This is copied from Gutenberg's approach but we have more granular matching of specific packages we want to size check and we also have a 100B threshold which feels sensible to me to keep noise down and focus on impactful changes but I'm open to discussion on that one.

We also introduce a "distclean" command that will blow away node_modules, build folders etc between builds.

I did some experimenting and decided to exclude the wc-admin build, because its almost all a duplicate of the packages build. If in future someone wants to measure a specific wc-admin built asset they can adjust the configuration. I found the `exclude` rule is limited and does not function as I'd expect either (for example it would not ignore folders named `chunks` within a match) so it was simpler to be specific about includes and leave the default exclude.



### How to test the changes in this Pull Request:

I tested this quite simply and I'll screenshot the result here. You'll need to take my word for it or edit this PR yourself to confirm. Here's what I did:

1. Added an import to checkout block in woocommerce blocks of a large library I could think of (`Immutable`). Added some code too to ensure the library was not tree-shaken
2. Push the change and let the new CI check run
3. It posted the comment screenshotted below:


<img width="993" alt="Screenshot 2024-05-29 at 12 46 19 PM" src="https://github.com/woocommerce/woocommerce/assets/1281828/c558c6a2-77c1-4aa8-aee4-8ae12bd6919f">

As you can see a predictable result, the vendors increases for the introduction of immutable and the checkout increases for the new code.

And once the changes are reverted and branch is up to date with trunk, the comment is updated:

<img width="1050" alt="Screenshot 2024-05-29 at 12 59 29 PM" src="https://github.com/woocommerce/woocommerce/assets/1281828/f6f4f2a1-fd66-4339-8ab7-de34969d95e3">



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
